llvm/utils/TableGen/IntrinsicEmitter.cpp

```diff
@@ -481,9 +482,33 @@ static AttributeSet getIntrinsicArgAttributeSet(LLVMContext &C, unsigned ID) {
 static AttributeSet getIntrinsicFnAttributeSet(LLVMContext &C, unsigned ID) {
   switch (ID) {
     default: llvm_unreachable("Invalid attribute set number");)";
-  for (const CodeGenIntrinsic &Intrinsic : Ints) {
+
+  auto getEffectiveME = [](const CodeGenIntrinsic &Int) {
+    MemoryEffects ME = Int.ME;
+    // TODO: IntrHasSideEffects should affect not only readnone intrinsics.
+    if (ME.doesNotAccessMemory() && Int.hasSideEffects)
+      ME = MemoryEffects::unknown();
+    return ME;
+  };
+
+  // Returns true is `Int` has a non-empty set of function attributes.
+  // Note that NoUnwind = !canThrow, so we need to negate its sense for testing
+  // if the intrinsic has NoUnwind attribute.
+  auto hasFnAttributes = [&getEffectiveME](const CodeGenIntrinsic &Int) {
```

Comment:
These don't need to be lambdas, they can just be deeper functions

Reply:
Done. Also extracted another lambda I had added earlier into a function.